ghcup

```diff
@@ -994,7 +994,9 @@ while [ $# -gt 0 ] ; do
            if [ "${TARGET_LOCATION}" ] ; then
                self_update "${TARGET_LOCATION}"
            else
-               self_update "${HOME}/.local/bin"
+               # https://stackoverflow.com/a/246128/1041691
+               DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
```

Comment:
This is bash syntax, we are not using bash. Instead of `"${BASH_SOURCE[0]}"` you can probably just use `"$0"`.

Check https://gist.github.com/tvlooy/cbfbdb111a4ebad8b93e for tests and discussions of different approaches. This one is also not entirely safe. Mind that we are replacing a file here, so it should be safe.

Reply:
```
# @VARIABLE: SOURCE
# @DESCRIPTION:
# The $0 argument, which contains
# the script name.
SOURCE="$0"


# @FUNCTION: script_dir
# @DESCRIPTION:
# Portably gets the full directory of where
# this script resides in and prints it to stdout.
# @STDOUT: script directory
script_dir() {
    mysource=${SOURCE}

    while [ -h "${mysource}" ]; do
        DIR="$( cd -P "$( dirname "${mysource}" )" && pwd )"
        mysource="$(readlink "${mysource}")"
        [ "${mysource%${mysource#?}}"x != '/x' ] && mysource="${DIR}/${mysource}"
    done
    DIR="$( cd -P "$( dirname "${mysource}" )" && pwd )"
    echo "${DIR}"

    unset mysource DIR
}
```

I think this will do and is posix compliant and portable.